pennylane/gradients/general_shift_rules.py

```diff
@@ -391,28 +391,41 @@ def generate_shifted_tapes(tape, index, shifts, multipliers=None):
             The length should match the one of ``shifts``. Each multiplier scales the
             corresponding gate parameter before the shift is applied. If not provided, the
             parameters will not be scaled.
+        broadcast (bool): Whether or not to use broadcasting to create a single tape
+            with the shifted parameters.
 
     Returns:
         list[QuantumTape]: List of quantum tapes. In each tape the parameter indicated
             by ``index`` has been shifted by the values in ``shifts``. The number of tapes
-            matches the lenth of ``shifts`` and ``multipliers`` (if provided).
+            matches the length of ``shifts`` and ``multipliers`` (if provided).
+            If ``broadcast=True`` was used, the list contains a single broadcasted tape
+            with all shifts distributed over the broadcasting dimension. In this case,
+            the ``batch_size`` of the returned tape matches the length of ``shifts``.
     """
-    params = list(tape.get_parameters())
-    if multipliers is None:
-        multipliers = np.ones_like(shifts)
 
-    tapes = []
-
-    for shift, multiplier in zip(shifts, multipliers):
+    def _copy_and_shift_params(tape, params, idx, shift, mult):
+        """Create a copy of a tape and of parameters, and set the new tape to the parameters
+        rescaled and shifted as indicated by ``idx``, ``mult`` and ``shift``."""
         new_params = params.copy()
-        shifted_tape = tape.copy(copy_operations=True)
-        new_params[index] = new_params[index] * qml.math.convert_like(multiplier, new_params[index])
-        new_params[index] = new_params[index] + qml.math.convert_like(shift, new_params[index])
+        new_params[idx] = new_params[idx] * qml.math.convert_like(
+            mult, new_params[idx]
+        ) + qml.math.convert_like(shift, new_params[idx])
 
+        shifted_tape = tape.copy(copy_operations=True)
         shifted_tape.set_parameters(new_params)

```

Comment:
While maybe not applicable here, I think this copy and set approach needs to become a construct-with-new-params option instead.

Reply:
I agree. Do I get this right, that this is pointing towards the tape/circuit data refactor? :)

Reply:
Yep, that's right. Nothing to be done here, but worth a mention just for future visibility on the topic.